### PR TITLE
First pull request

### PR DESCRIPTION
**Pull request by RodrigoJPT**
log in, requirement export, and template download all working.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -22,115 +20,143 @@ var newRequirement = {
...
+    $.ajax({
+      method: "GET",
+      crossDomain: true,
+      url: userInfo.spiraUrl
+      + "services/v5_0/RestService.svc/projects?username="
+      + userInfo.username
+      + "&api-key="
+      + userInfo.apikey,
+      success: function (data, textStatus, response) {
+        console.log(response.status);
```
Comment (BraffZackland) [inline app.js:67]: maybe these should be put into their own functions for readability?

File: app.js [hunk 2]
```diff
@@ -22,115 +20,143 @@ var newRequirement = {
...
+      dataType: "json",
+      url: userInfo.spiraUrl
+      + "services/v5_0/RestService.svc/projects/"
+      + $('#projects').val()
+      + "/requirements?username="
+      + userInfo.username
+      + "&api-key="
+      + userInfo.apikey,
+      data: JSON.stringify(req)
+    }).done(function (data) {
```
Comment (BraffZackland) [inline app.js:138]: can these ajax functions be streamlined a bit to make them easier to read?

File: app.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    })
+  };
+
+  function logIn() {
+    //userInfo.spiraUrl = $("#url").val();
+    //userInfo.username = $("#username").val();
+    //userInfo.apikey = $("#apikey").val();
+
+    if (userInfo.spiraUrl.charAt(userInfo.spiraUrl.length - 1) != "/") {
+      userInfo.spiraUrl += "/";
```
Comment (simonhbor) [inline app.js:42]: good check!!

File: app.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        .then(function () {
+          getReqValues(sheetRange.values.length);
+        })
+    });
+  }
+
+  function getReqValues(rows) {
+    return Excel.run(function (context) {
+      const sheet = context.workbook.worksheets.getActiveWorksheet();
+      const inputRange = "A3:K" + rows;
```
Comment (simonhbor) [inline app.js:101]: a 'magic number' that could be made into a param/variable

File: app.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      const inputValues = sheet.getRange(inputRange);
+      inputValues.load();
+      return context.sync()
+        .then(function () {
+          buildRequirementObject(inputValues.values);
+        })
+    });
+  }
+
+  function buildRequirementObject(pulledValues) {
```
Comment (simonhbor) [inline app.js:111]: pedantic point but this buildreq function both builds and then sends - could be split into two functions

**Review by simonhbor — APPROVED** [review]
Looks good - just about understood what is going on ;)

**Pull request merged**